workflow/controller/retry_tweak_test.go

```diff
@@ -59,6 +59,24 @@ func TestFindRetryNode(t *testing.T) {
 			Children:     []string{},
 			TemplateName: "tmpl2",
 		},
+		"E1": wfv1.NodeStatus{
+			ID:         "E1",
+			Type:       wfv1.NodeTypeRetry,
+			Phase:      wfv1.NodeRunning,
+			BoundaryID: "A1",
+			Children:   []string{},
+			TemplateRef: &wfv1.TemplateRef{
+				Name: "tmpl1",
```

Comment:
shouldn't this have a `template` field as well? the test is passing with only `name`?

Reply:
Ok,I have added it.

Reply:
> the test is passing with only `name`?

Oh I guess the test passed bc `nil == nil`. The code checks whether the `templateRef` struct isn't `nil`, but not the individual fields.
They should never be `nil` in a real use-case, so I suppose that is a test edge-case